packages/components/src/components/Dropzone/Dropzone.tsx

```diff
@@ -0,0 +1,252 @@
+};
+
+const Dropzone = React.forwardRef<HTMLDivElement, DropzoneProps>(
+  (
+    {
+      fileTypes = { "application/pdf": [".pdf"] },
+      cancelDocumentUpload,
+      sendDocument,
+    },
+    ref
+  ) => {
+    const [status, setStatus] = useState("default");
+    const [progress, setProgress] = useState(0);
+    const [errors, setErrors] = useState("");
+    const {
+      acceptedFiles,
+      getRootProps,
+      getInputProps,
+      isDragActive,
+      isDragAccept,
+      isDragReject,
+      isFocused,
+    } = useDropzone({
+      maxFiles: 1,
+      maxSize: MAX_FILE_SIZE,
+      multiple: false,
+      accept: {
+        ...fileTypes,
+      },
+      onDropAccepted: () => {
+        setStatus("loading");
+        sendDocument(acceptedFiles[0], (progress) => {
+          setProgress(progress);
+        })
+          .then(() => {
+            setProgress(0);
+            setStatus("success");
+          })
+          .catch(() => {
+            setStatus("uploadError");
+          });
+      },
+      onDropRejected: (fileRejections) => {
+        setStatus("error");
+        forEach(fileRejections, (file) => {
+          forEach(file.errors, (err) => {
+            if (err.code === "file-too-large") {
+              setErrors("File must be less than 50mb.");
+            }
+            if (err.code === "file-invalid-type") {
+              setErrors("Wrong file type. PDF format only.");
+            }
+          });
+        });
+      },
+    });
+
+    React.useEffect(() => {
+      if (isDragActive) {
+        setStatus("default");
+      }
+    }, [isDragActive]);
+
+    return (
+      <>
+        <Box.div
+          {...getVariants(status, isDragAccept, isDragReject, isFocused)}
+          alignItems="center"
+          borderRadius={"borderRadius30"}
+          borderStyle={"borderDashed"}
+          borderWidth={"borderWidth10"}
+          display="flex"
+          flexDirection="column"
+          padding={"space70"}
+          {...getRootProps({
+            status,
+            isDragActive,
+            isDragAccept,
+            isDragReject,
+          })}
+          ref={ref}
+        >
+          <input {...getInputProps()} />
+          {getIcon(status, isDragActive, isDragAccept, isDragReject)}
+
+          {!isDragActive && status === "loading" && (
+            <>
+              {acceptedFiles[0].name}
+              <Box.div
+                alignItems="center"
+                display="flex"
+                marginTop="space40"
+                maxWidth="300px"
+                w="100%"
+              >
+                <Box.div maxWidth="280px" w="100%">
+                  <ProgressBar value={progress} />
+                </Box.div>
+                <CancelUploadButton
+                  onClick={(event) => {
+                    event.stopPropagation();
+                    cancelDocumentUpload(acceptedFiles[0].name).then(() =>
```

Comment:
Could we upload more than one file? What happens here if we try to add more files?

Reply:
V1 only allows the uploading of one file. V2 we might accept multiple files. In that case we'll have to map over the acceptedFiles array and sendDocument on each file. Make sense?